lib/Mojo/DOM/CSS.pm

```diff
@@ -173,18 +215,28 @@ sub _namespace {
   return !length $ns;
 }
 
+sub _parent {
+  my ($tree) = @_;
+  return $tree->[3] if $tree->[0] eq 'tag';
+  return $tree->[2] if $tree->[0] eq 'text';
+  return undef      if $tree->[0] eq 'root';
```

Comment:
This conditional makes no sense.

Reply:
I am unsure what you mean by this.

`_parent` is intended as a general utility function, that given a node traverses one step up in the syntax tree, and explicitly handles the 3 possible cases as I understand them.

Reply:
Take a closer look, what happens when `if $tree->[0] eq 'root'` doesn't match? 😉